src/Raven.Server/Documents/DocumentIdWorker.cs

```diff
@@ -230,11 +236,19 @@ private static ByteStringContext.InternalScope UnlikelyGetLowerUnicode(ByteStrin
 
             var writePos = ptr + maxStrSize;
 
+            var sizeDifference = idSize - JsonParserState.VariableSizeIntSize(strLength);
+            if (sizeDifference > 0)
```

Comment:
Can this be negative?

Reply:
The strLength should be smaller/equal to the initial value that was used to calculate idSize, so the difference should be positive/zero.

Reply:
added debug assert there

Reply:
So we can just have an assert here, and avoid the `if` entirely?

Reply:
no, we should move the pointer for idSize if there is difference, this is the same code like here :
https://github.com/ravendb/ravendb/blob/v4.2/src/Raven.Server/Documents/DocumentIdWorker.cs#L285